src/gsts3sink.c

```diff
@@ -105,7 +105,7 @@ gst_s3_sink_class_init (GstS3SinkClass * klass)
           G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY | G_PARAM_STATIC_STRINGS));
 
   g_object_class_install_property (gobject_class, PROP_KEY,
-      g_param_spec_string ("key", "S3 key",
+      g_param_spec_string ("location", "S3 key",
```

Comment:
I agree with you that we should have a location as a parameter, but I don't think it should be equivalent to the S3 key. What do you think about having 2 ways of defining the pipeline destination:
* `location`, which is an URI, e.g. `s3://bucket-name/path/to/key`
* `key` and `bucket` pair stays as it was

We could perform validation on setting the properties to make sure that only one way of setting the destination is used by the user.

That would also enable us to implement GstUriHandler for `s3` protocol.

Reply:
I totally agree that `location` should be a full S3 URI.

Can't see why `key` and `bucket` should stay as it was in that case? Perhaps it makes sense as a getter property for some sort of validation check, but having both location and bucket+key pair may be confusing. Correct me if I'm wrong.

Reply:
It'd be good to keep `key` and `bucket` for backward compatibility reasons - we could mark them as deprecated with a warning message about alternative approach, and then remove in the future.